### PR TITLE
[RW-1791][risk=no] Complete/clean-up modifiers validation

### PR DESCRIPTION
**Pull request by dolbeew**
The main purpose here is to make validation consistent across the app. Changing the modifiers page to act like other forms was the quickest fix, but we can go the other way if needed. Also fixed several bugs in modifiers found along the way

### CODE REVIEW

File: ui/src/app/cohort-search/modifier-page/modifier-page.component.ts [hunk 1]
```diff
@@ -319,22 +342,10 @@ export class ModifierPageComponent implements OnInit, OnDestroy, AfterContentChe
...
-  negativeNumber() {
-   this.modifiers$.forEach(item => {
-    const modArr = item.map(modValue => {
-       return modValue.toJS().operands.map( o => {
-          return o < 0;
-        });
-      });
-     this.showError = modArr.toJS().flat().includes(true);
-    });
+  get disableCalculate() {
```
Comment (freemabd1) [inline ui/src/app/cohort-search/modifier-page/modifier-page.component.ts]: Please disable the Finish button as well.

File: ui/src/app/cohort-search/modifier-page/modifier-page.component.html [hunk 2]
```diff
@@ -25,13 +25,13 @@
     <div *ngFor="let mod of modifiers; let i = index" [formGroupName]="mod.name">
       <div class="label-flex">
         <label class="modifier-event-header" [for]="mod.name">{{ mod.label }}</label>
```
Comment (freemabd1) [inline ui/src/app/cohort-search/modifier-page/modifier-page.component.html]: I'm able to add assas as a date and i don't get a validation message. I'm able to click Calculate and i get an error message

File: ui/src/app/cohort-search/modifier-page/modifier-page.component.html [hunk 3]
```diff
@@ -100,11 +100,11 @@
...
-        class="btn "
+      <button type="button"
+        class="btn calculate-btn"
         [clrLoading]="preview.get('requesting')"
         (click)="requestPreview()"
-        [disabled] = "!formChanges">
+        [disabled]="disableCalculate">
         Calculate
       </button>
     </div>
```
Comment (freemabd1) [inline ui/src/app/cohort-search/modifier-page/modifier-page.component.html]: Also, we need to limit the integers that can be entered in the age at event field and number of occurrences as well. I was able to enter 100000000000000000000000000000000000000

File: ui/src/app/cohort-search/modifier-page/modifier-page.component.ts [hunk 4]
```diff
@@ -110,7 +119,18 @@ export class ModifierPageComponent implements OnInit, OnDestroy, AfterContentChe
...
-  showError = false;
+  errors = new Set();
+  readonly errorMessages = {
+    ageAtEvent: {
+      range: 'Values for Age At Event must be between 1 and 120',
+      integer: 'Values for Age At Event must be valid integers'
+    },
+    hasOccurrences: {
+      range: 'Values for Has Occurrences must be between 1 and 99',
+      integer: 'Values for Has Occurrencest must be valid integers'
```
Comment (freemabd1) [inline ui/src/app/cohort-search/modifier-page/modifier-page.component.ts]: typo: please remove the t from Occurrencest

**Comment by dolbeew** [comment]
Should have addressed all 4 points from the last comment

File: ui/src/app/cohort-search/modifier-page/modifier-page.component.ts (hunk 4 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/modifier-page/modifier-page.component.ts]: Done

**Pull request merged**